### PR TITLE
Added code to handle timer for unregistered UE and imsi is not known

### PR DESCRIPTION
**Pull request by rsarwad**
## Summary
When UE performs GUTI based attach, mme sends Identity Request and mean while mme goes down, UE contexts are not updated to DB as imsi is not yet known to mme. But on arrival of Attach request in Initial UE message, there is an entry in enb_ue_s1ap_id_ue_context_htbl with key as enb_s1ap_id_key and data as mme_ue_s1ap_id. As result of this, on mme recovery UE contexts are not retrieved but content of hashlist, enb_ue_s1ap_id_ue_context_htbl  would be retrieved. So there is an entry to this list as enb_s1ap_id_key
So when new next attach request comes with same enb_ie_s1ap_id as before then this attach request would get dropped as enb_s1ap_id_key already exists in the hashlist, enb_ue_s1ap_id_ue_context_htbl.

So new attach request arrival, if enb_ue_s1ap_id key exists, s1ap shall send an indication to mme_app to delete UE contexts
## Test Plan
Executed s1ap sanity test suite
New test cases are scripted to validate the 3470 timer and 3450 timer while MME restarts, are available in PR, https://github.com/magma/magma/pull/4139

### CODE REVIEW

File: lte/gateway/c/oai/tasks/mme_app/mme_app_context.c [hunk 1]
```diff
@@ -2478,3 +2478,36 @@ static void mme_app_handle_timer_for_unregistered_ue(
...
+
+void mme_app_remove_stale_ue_context(
+    mme_app_desc_t* mme_app_desc_p,
+    itti_s1ap_remove_stale_ue_context_t* s1ap_remove_stale_ue_context) {
+  OAILOG_FUNC_IN(LOG_MME_APP);
+  enb_s1ap_id_key_t enb_s1ap_id_key = INVALID_ENB_UE_S1AP_ID_KEY;
+  MME_APP_ENB_S1AP_ID_KEY(
+      enb_s1ap_id_key, s1ap_remove_stale_ue_context->enb_id,
+      s1ap_remove_stale_ue_context->enb_ue_s1ap_id);
+  if (INVALID_ENB_UE_S1AP_ID_KEY != enb_s1ap_id_key) {
```
Comment (ssanadhya) [inline lte/gateway/c/oai/tasks/mme_app/mme_app_context.c]:
```suggestion
  if (enb_s1ap_id_key == INVALID_ENB_UE_S1AP_ID_KEY) {
   OAILOG_FUNC_OUT(LOG_MME_APP);
   }
```

File: lte/gateway/c/oai/tasks/mme_app/mme_app_context.c [hunk 2]
```diff
@@ -2478,3 +2478,36 @@ static void mme_app_handle_timer_for_unregistered_ue(
...
+    uint64_t mme_ue_s1ap_id = INVALID_MME_UE_S1AP_ID;
+    if (hashtable_uint64_ts_get(
+            mme_app_desc_p->mme_ue_contexts.enb_ue_s1ap_id_ue_context_htbl,
+            (const hash_key_t) enb_s1ap_id_key,
+            &mme_ue_s1ap_id) == HASH_TABLE_OK) {
+      ue_mm_context_t* ue_context_p =
+          mme_ue_context_exists_mme_ue_s1ap_id(mme_ue_s1ap_id);
+      if (!ue_context_p) {
```
Comment (ssanadhya) [inline lte/gateway/c/oai/tasks/mme_app/mme_app_context.c]: Add a comment saying that if ue_context does not exist for the mme_ue_s1ap_id, delete the corresponding enb_ue_s1ap_id entry.

**Review by ssanadhya — APPROVED** [review]
Looks good, with minor comments

**Comment by 119Vik** [comment]
RestartCWAGTests

**Comment by 119Vik** [comment]
RestartCWAGTests

**Review by ssanadhya — APPROVED** [review]

**Pull request merged**